### PR TITLE
ipopt 3.12.12 (new formula)

### PR DESCRIPTION
**Pull request by jamiesnape**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/ipopt.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  def install
+    ENV.delete("MPICC")
+    ENV.delete("MPICXX")
+    ENV.delete("MPIFC")
+    ENV.deparallelize
+    ENV.fortran
+
+    chdir buildpath/"ThirdParty/Mumps" do
+      system "./get.Mumps"
```
Comment (fxcoudert) [inline Formula/ipopt.rb]: This is not how we handle dependencies in Homebrew. You will need to provide `mumps` as a separate formula, or to vendor it inside this formula (as a `resource` block).

File: Formula/ipopt.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sha256 "7baeb713ef8d1999bed397b938e9654b38ad536406634384455372dd7e4ed61f"
+  head "https://github.com/coin-or/Ipopt.git"
+
+  depends_on "gcc"
+
+  def install
+    ENV.delete("MPICC")
+    ENV.delete("MPICXX")
+    ENV.delete("MPIFC")
+    ENV.deparallelize
```
Comment (fxcoudert) [inline Formula/ipopt.rb]: Why is parallel build not supported? Is this a known bug?

File: Formula/ipopt.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ENV.fortran
```
Comment (fxcoudert) [inline Formula/ipopt.rb]: We don't use `ENV.fortran` anymore.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ipopt.rb (hunk 1 above)
Comment (jamiesnape) [reply Formula/ipopt.rb]: I know, but the MUMPS Makefile is in such bad shape, there is not a responsive upstream, and the IPOPT mangles the library name so it is not going to collide. Basically, this is going to be the only way to get IPOPT into Homebrew (search closed pull requests for a refresher).

File: Formula/ipopt.rb (hunk 2 above)
Comment (jamiesnape) [reply Formula/ipopt.rb]: Technically, parallel install is not supported. I can probably get the build to work in parallel.

File: Formula/ipopt.rb (hunk 1 above)
Comment (jamiesnape) [reply Formula/ipopt.rb]: Would using a resource block to get MUMPS and the build files in IPOPT be acceptable?

Comment (fxcoudert) [reply Formula/ipopt.rb]: Let's try it that way, yes.

Comment (jamiesnape) [reply Formula/ipopt.rb]: Done.

File: Formula/ipopt.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def install
+    ENV.delete("MPICC")
+    ENV.delete("MPICXX")
+    ENV.delete("MPIFC")
+
+    resource("coinmumps").stage do
+      # https://github.com/coin-or-tools/ThirdParty-Mumps/blob/35dba763084f6f9ed4db087e2e5e53e80a92bb15/get.Mumps#L51-L53
+      system "patch", "-p1", "-i", buildpath/"ThirdParty/Mumps/mumps.patch"
```
Comment (fxcoudert) [inline Formula/ipopt.rb]: So it actually needs a patched, old and unmaintained version? Not sure if that's cool for Homebrew core. What about maintaining the whole thing in a tap?

Comment (jamiesnape) [reply Formula/ipopt.rb]: Trying to migrate from a tap (one of the many for this exact formula, which is why, technical issues aside, it feels like it should be in core).

It doesn't actually need an old MUMPS, but the build files are written for that version. We could use a new MUMPS, but then we have to deal with the MUMPS build files. Let me see if that is viable.

Comment (jamiesnape) [reply Formula/ipopt.rb]: @fxcoudert PTAL. This uses the latest MUMPS.

**Review by SMillerDev — APPROVED** [review]

**Comment by jamiesnape** [comment]
It would be good to get a definite yes/no on this. I think there is a lot of demand for this formula, but obviously there are some issues with the MUMPS dependency.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
OK @jamiesnape let's go with this, I'm not entirely satisfied with the state of the MUMPS building… is there any way you can contribute to upstream improvements to streamline this for future versions?

**Comment by jamiesnape** [comment]
I can certainly try, but the development process seems to be pretty much internal/undocumented.

**Pull request closed without merging**